resources/deployments/model/deployments.go

```diff
@@ -194,18 +302,26 @@ func (d *DeploymentsModel) GetDeploymentForDeviceWithCurrent(ctx context.Context
 		return nil, nil
 	}
 
-	link, err := d.imageLinker.GetRequest(ctx, deployment.Image.Id,
+	if err := d.assignArtifact(ctx, deployment, deviceDeployment, installed); err != nil {
```

Comment:
I'm not sure if this is ok to assign a new image each time we reach this place. I'd guess that once an image is assigned you could skip looking up a new image. For instance suppose that devices are retrying image download by first asking to see if there is a deployment. This would make sense if device type can change in between the requests. @estenberg @maciejmrowiec is this possible?